### PR TITLE
WIP: Informer and RBAC linting

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

File: cmd/internal/codegen/parse/parser.go [hunk 1]
```diff
@@ -62,11 +65,53 @@ func NewAPIs(context *generator.Context, arguments *args.GeneratorArgs) *APIs {
...
+				if g == "*" || group == gvk.Group {
+					groupFound = true
+					break
+				}
+				if (group == "core" || group == "") && (gvk.Group == "core" || gvk.Group == "") {
+					groupFound = true
+					break
+				}
+			}
+			if !groupFound {
```
Comment (droot) [inline cmd/internal/codegen/parse/parser.go]: Quick q: Don't we have to search in all the rules before declaring that group is not found ?

File: cmd/internal/codegen/parse/parser.go [hunk 2]
```diff
@@ -62,11 +65,53 @@ func NewAPIs(context *generator.Context, arguments *args.GeneratorArgs) *APIs {
 	b.parseControllers()
 	b.parseRBAC()
 	b.parseInformers()
+	b.checkRbac()
 	b.parseAPIs()
 	b.parseCRDs()
 	return b
 }
 
+func (b *APIs) checkRbac() {
```
Comment (droot) [inline cmd/internal/codegen/parse/parser.go]: Rename it to reflect that it is validating RBAC annotations are matching Informers annotation ?
A few ugly suggestions :) 
`validateAnnotations`
`validateRBACandInformersAnnotations`
`EnsureRBACmatchesInformerAnnotations`

File: cmd/internal/codegen/parse/parser.go (hunk 1 above)
Comment (pwittrock) [reply cmd/internal/codegen/parse/parser.go]: Good catch

File: cmd/internal/codegen/parse/parser.go (hunk 2 above)
Comment (pwittrock) [reply cmd/internal/codegen/parse/parser.go]: Named it `validateRBACAnnotations`

**Comment by pwittrock** [comment]
@droot Thoughts on how best to test this?

**Comment by droot** [comment]
@pwittrock test in travis is catching a real issue of missing RBAC annotations :)
The code change is perfect candidate for unit test, but doing so in current code structure is a bit tough. I am taking another look at the parse package to see if we can make it unit-test-friendly.

**Comment by droot** [comment]
@pwittrock FYI. I have cloned this PR locally and adding unit-test for it.

**Pull request closed without merging**